packages/react-charting/src/components/GaugeChart/GaugeChart.types.ts

```diff
@@ -0,0 +1,252 @@
+import { IStyle, ITheme } from '@fluentui/react/lib/Styling';
+import { IStyleFunctionOrObject } from '@fluentui/react/lib/Utilities';
+import { ILegendsProps } from '../Legends/index';
+import { IAccessibilityProps } from '../../types/index';
+import { ICalloutProps } from '@fluentui/react/lib/Callout';
+
+export interface IGaugeChartSegment {
+  /**
+   * Legend text for a segment
+   */
+  legend: string;
+
+  /**
+   * Size of the segment
+   */
+  size: number;
+
+  /**
+   * Color of the segment. If not provided, a placeholder color will be used.
+   */
+  color?: string;
+
+  /**
+   * Accessibility data for the segment
+   */
+  accessibilityData?: IAccessibilityProps;
+}
+
+export enum GaugeValueFormat {
+  Percentage = 'percentage',
+  Fraction = 'fraction',
+}
+
+export interface IGaugeChartProps {
+  /**
+   * Width of the chart
+   */
+  width?: number;
+
+  /**
+   * Height of the chart
+   */
+  height?: number;
+
+  /**
+   * Title of the chart
+   */
+  chartTitle?: string;
+
+  /**
+   * Current value of the gauge
+   */
+  chartValue: number;
+
+  /**
+   * Sections of the gauge
+   */
+  segments: IGaugeChartSegment[];
+
+  /**
+   * Minimum value of the gauge
+   */
+  minValue?: number;
+
+  /**
+   * Maximum value of the gauge
+   */
+  maxValue?: number;
+
+  /**
+   * Additional text to display below the chart value
+   */
+  sublabel?: string;
+
+  /**
+   * Hide the min and max values of the gauge
+   * @defaultvalue false
+   */
+  hideLimits?: boolean;
+
+  /**
+   * Format of the chart value
+   * @defaultvalue GaugeValueFormat.Percentage
+   */
+  chartValueFormat?: GaugeValueFormat;
+
+  /**
+   * Decides wether to show/hide legends
+   * @defaultvalue false
+   */
+  hideLegend?: boolean;
+
+  /*
+   * Props for the legends in the chart
+   */
+  legendProps?: Partial<ILegendsProps>;
+
+  /**
+   * Do not show tooltips in chart
+   * @defaultvalue false
+   */
+  hideTooltip?: boolean;
+
+  /**
+   * Additional CSS class(es) to apply to the chart
+   */
+  className?: string;
+
+  /**
+   * Theme (provided through customization)
+   */
+  theme?: ITheme;
+
+  /**
+   * Call to provide customized styling that will layer on top of the variant rules
+   */
+  styles?: IStyleFunctionOrObject<IGaugeChartStyleProps, IGaugeChartStyles>;
+
+  /**
+   * Prop to define the culture to localize the numbers and dates
+   */
+  culture?: string;
+
+  /**
+   * Props for the callout in the chart
+   */
+  calloutProps?: Partial<ICalloutProps>;
+}
+
+export interface IGaugeChartStyleProps {
+  /**
+   * Theme (provided through customization)
+   */
+  theme: ITheme;
+
+  /**
+   * Font size of the chart value
+   */
+  chartValueSize?: number;
+
+  /**
+   * Width of the chart
+   */
+  width?: number;
+
+  /**
+   * Height of the chart
+   */
+  height?: number;
+
+  /**
+   * Additional CSS class(es) to apply to the chart
+   */
+  className?: string;
+
+  /**
+   * Color of the line
+   */
+  lineColor?: string;
```

Comment:
which line is this?

Reply:
This is for the legend line in hover card